src/front/wgsl.rs

```diff
@@ -7,7 +7,7 @@ use crate::{
     FastHashMap,
 };
 
-#[derive(Debug, PartialEq)]
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
why do we need Copy+Clone here?

Reply:
We don't quite "need" it. It just made the loop in the test a tiny bit simpler. Without it the line `if result == Token::End` would result in a compile error because the value of `result` was moved into the Vec in the line before.

I can instead check `if results.last() == Some(&Token::End)`,if you want.